server/src/main/java/org/server/SupServer.java

```diff
@@ -2,90 +2,114 @@
+        String clientname = "";
+        Socket sock;
+
+        public SupClientHandler(Socket clientSocket) {
+            try {
+                sock = clientSocket;
+                InputStreamReader isReader = new InputStreamReader(sock.getInputStream());
+                reader = new BufferedReader(isReader);
+            }
+            catch(Exception ex) {
+                ex.printStackTrace();
+            }
+        }
+
+        public void run() {
+            // Run until connection is broken or user terminates
+            int character = 0;
+            try {
+                while((character = reader.read()) != -1) {
+
+                    String message = "";
+
+                    // Go until EOT character, end of transmission
+                    while(character != 0x04) {
+                        message = message + Character.toString((char) character);
+                        character = reader.read();
+                    }
+
+                    // now we have the message read in
+                    System.out.println("read \"" + message + "\" from " + Thread.currentThread().getName());
+
+                    // at this point, for the first message only, we'll need to parse the message
+                    // for a username and add it to contact list, like:
+                    // Only the first time expect this!
+                    if(clientname.isEmpty()) {
+                        String requestedName = message.split(" ")[1];
+                        if (Contacts.getInstance().hasContact(requestedName)) {
+                            // TODO return error, username taken,end connection
+                            System.out.println("Contact name taken: " + requestedName );
+                        }
+                        else {
+                            clientname = requestedName;
+                            Contacts.getInstance().addContact( clientname, new PrintWriter(sock.getOutputStream()) );
+                            System.out.println("New contact name: " + clientname );
+                        }
+                    }
+
+                    // for all chat messages (and only chat messages), send to the destination
+                    String toname = "TODO"; // need to parse that out of the chat message
+                    tellSomeone(message, this.clientname, toname );
+                }
+
+
+            } catch (IOException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+            // if they were actually authenticated and left, log them off..
+            if(!clientname.isEmpty()) {
+                System.out.println(clientname + " is no longer online.");
+                // remove user from active contacts
+                Contacts.getInstance().removeContact(clientname);
+            }
         }
-      }
-      catch(Exception ex) {
-        ex.printStackTrace();
-      }
-    }
-  }
-  
-  public void start () {
```

Comment:
these didn't really change, i just replaced tabs with 4 spaces to match our coding standard. probably should have done that in a different commit...